### PR TITLE
Reimplement CUDNN_jll using platform tags

### PR DESCRIPTION
**Pull request by maleadt**
Building on https://github.com/JuliaPackaging/Yggdrasil/pull/4677, this is a first test of a JLL that actually uses the JLL machinery to depend on and load a compatible set of CUDA runtime libraries.

### CODE REVIEW

**Comment by maleadt** [comment]
Is Yggdrasil's clone of General lagging behind? 

```
 Warning: Unable to resolve CUDA_Runtime_jll
└ @ BinaryBuilderBase /depot/packages/BinaryBuilderBase/BstWr/src/Dependencies.jl:228
```

Works locally.

**Comment by giordano** [comment]
https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=18613&view=logs&j=96c13d9d-dfd2-51e5-2f76-4a0b3ce09abc&t=09532434-8b0a-5f0e-118d-a3181cde5ec8&l=26
```
[ Info: PkgServer: https://us-east.pkg.julialang.org/registries.conservative -- last updated: Thu, 24 Mar 2022 09:05:22 GMT (3 hours, 12 minutes, 23 seconds ago)
┌ Warning: The PkgServer registry is older than 1 hour
└ @ Main none:27
```
It's older than the latest commit to General, but newer than the commit which included `CUDA_Runtime_jll` :thinking:

File: C/CUDNN/build_tarballs.jl [hunk 1]
```diff
@@ -49,29 +31,59 @@ elif [[ ${target} == x86_64-w64-mingw32 ]]; then
...
+
         if platform == Platform("x86_64", "windows")
             push!(sources,
                 ArchiveSource("http://www.winimage.com/zLibDll/zlib123dllx64.zip",
                               "fd324c6923aa4f45a60413665e0b68bb34a7779d0861849e02d2711ff8efb9a4"))
         end
-        build_tarballs(ARGS, name, version, sources, script, [augmented_platform],
-                       products, dependencies; lazy_artifacts=true)
+
+        dependencies = [Dependency(PackageSpec(name="CUDA_Runtime_jll");
```
[the remark this answers is not in the document]
Comment (giordano) [reply C/CUDNN/build_tarballs.jl:66]: Like this:
```suggestion
        dependencies = [Dependency(PackageSpec(name="CUDA_Runtime_jll", uuid="76a88914-d11a-5bdc-97e0-2f5a05c973a2");
```

[the remark this answers is not in the document]
Comment (giordano) [reply C/CUDNN/build_tarballs.jl]: That did the trick :slightly_smiling_face:

**Comment by maleadt** [comment]
https://github.com/JuliaPackaging/Yggdrasil/pull/5699

**Pull request closed without merging**